File: assurance/fixtures/sdlc/mixed/node/tests/verify.mjs
import { readFile } from "node:fs/promises";
import kleur from "kleur";

kleur.enabled = true;
const expected = "\u001b[1m3.10:xn--fa-hia.de\n\u001b[22m";
const actual = await readFile("generated/value.txt", "utf8");
if (actual !== expected) {
  throw new Error("prepared mixed Node service output does not match its input");
}

File: assurance/fixtures/sdlc/mixed/python/tests/verify.py
from pathlib import Path

import idna

root = Path(__file__).resolve().parents[1]
if idna.__version__ != "3.10" or (root / "generated/value.txt").read_text() != "3.10:xn--fa-hia.de\n":
    raise SystemExit("prepared mixed Python service output does not match its input")
